src/WebJobs.Script/Rpc/LanguageWorkerProcessManager.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Script.Abstractions;
+using Microsoft.Azure.WebJobs.Script.Eventing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public class LanguageWorkerProcessManager : ILanguageWorkerProcessManager
+    {
+        private readonly IWorkerProcessFactory _processFactory;
+        private readonly IEnumerable<WorkerConfig> _workerConfigs = null;
+        private readonly IProcessRegistry _processRegistry;
+        private readonly ILogger _logger = null;
+        private readonly ILoggerFactory _loggerFactory = null;
+        private readonly IScriptEventManager _eventManager = null;
+        private readonly IRpcServer _rpcServer = null;
+        private readonly ILanguageWorkerConsoleLogSource _consoleLogSource;
+
+        public LanguageWorkerProcessManager(IRpcServer rpcServer,
```

Comment:
This name is a little confusing. This type now is simply a factory (it doesn't manage the language worker process), the name should reflect that, but it's confusing that we have another type called `DefaultWorkerProcessFactory`. It's not clear by looking at the type names and APIs how they differ.

Reply:
I messed up my commits while rebasing :( Fixing it now.